kernel/rule.mli

```diff
@@ -42,13 +41,15 @@ type rule_name =
   (** Rules of lambda pi modulo. The first parameter indicates whether
       the name of the rule has been given by the user. *)
 
+type condition = {left:term; right:term}
 
```

Comment:
Would it be hard to add a case to negate a conversion check? The constraint t1 != t2 would mean that the head-normal forms of t1 and t2 tell us that no substitution can make t1 and t2 convertible.

Reply:
Guards are not used during the type checking of a rule. But I have implemented a negation condition that is true if and only if the two terms are NOT convertible when Dedukti tries to apply the rule.